### PR TITLE
SEO improvements to guides 

### LINKED ISSUE #581: Replace underscores with dash in slugs (`_` -> `-`).

### LINKED ISSUE #582: Review guides to find opportunities for in-site cross-linking.
Where it makes sense for guides to link to one another, or to other parts of the site (such as the docs), we should do this.

### LINKED ISSUE #584: Add canonical links to cross-posted blog posts
I can't find the example we looked at, but I will.

### LINKED ISSUE #585: Review and edit guides into a listicle where appropriate.

### LINKED ISSUE #586: Review and edit guides to be easily scannable + parsable by readers (and google)
We need to ensure users can quickly find and consume the content they are interested in and to ensure that robots like it too. This means ensuring the purpose of the content is clear, ensuring that the content is concise, and easy to digest. 

- [ ] Add tldr style content at the top
- [ ] Ensure paragraphs are of an optimal length and that guides aren't too long, longer guides could be split into parts.
- [ ] Should we add some kind of CTAs in the content, follow on social media, contribute to the library, etc?

There are many articles like this but buffer talk about content + paragraph length later on. Buffer is a tool for publishing/ managing Social media channels etc so they should be relatively authoritative and the content seems to make sense. https://buffer.com/library/optimal-length-social-media/

### PR DESCRIPTION
**Pull request by aliabd**
Makes many SEO improvements to the guides

Closes #581, closes #582, closes #584, closes #585,  closes #586

### CODE REVIEW

**Comment by abidlabs** [comment]
Apologies for the late review @aliabd, but this looks great. Nice interlinking between docs and guides. I think we could do more links between the Hugging Face website and the Gradio website, which would probably boost our SEO significantly) but we can close the issue for now and keep it in mind as we write more guides in the future. 

Also thanks for making sure that the old links continue to work!

File: guides/01_getting-started/02_key-features.md [hunk 1]
```diff
@@ -1,19 +1,34 @@
 # Key Features
 
-Let's go through some of the most popular features of Gradio!
+Let's go through some of the most popular features of Gradio! 
+
+Here are Gradio's key features:
```
Comment (abidlabs) [inline guides/01_getting-started/02_key-features.md:5]: Can we shorten this? Suggestion:
```suggestion
Let's go through some key features you can add to your Gradio demo:
```

**Review by abidlabs — APPROVED** [review]

File: guides/01_getting-started/03_sharing-your-app.md [hunk 2]
```diff
@@ -1,5 +1,18 @@
 # Sharing Your App
 
+This guide will cover how sharing works with Gradio. 
+
+How to share your Gradio app:
```
[the remark this answers is not in the document]
Comment (aliabd) [reply guides/01_getting-started/03_sharing-your-app.md:5]: So what i'm trying out here and in the key features guide is mimicking the exact user query: How to share your Gradio app. (or trying to come as close as possible) So gonna keep this as is for now. I do agree that it's kind of annoyingly verbose and I do think that we should prioritize readability, etc over SEO but I really wanna run the experiment and see what it takes to for example show a listicle in results and how that affects search traffic. Will definitely come and clean things up after

**Comment by aliabd** [comment]
Thanks for the review @abidlabs, yup I reworded https://github.com/gradio-app/gradio/issues/583 to reflect this.

File: guides/01_getting-started/03_sharing-your-app.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (abidlabs) [reply guides/01_getting-started/03_sharing-your-app.md:5]: Can we still reduce it to one line? That way, there's less vertical space taken up before you actually start reading the Guide?

[the remark this answers is not in the document]
Comment (aliabd) [reply guides/01_getting-started/03_sharing-your-app.md]: Shortened them, thanks

**Pull request merged**